mappings/net/minecraft/entity/LivingEntity.mapping

```diff
@@ -169,23 +212,27 @@ CLASS net/minecraft/class_1309 net/minecraft/entity/LivingEntity
 	METHOD method_6059 hasStatusEffect (Lnet/minecraft/class_1291;)Z
 	METHOD method_6060 knockback (Lnet/minecraft/class_1309;)V
 		ARG 1 target
-	METHOD method_6062 cannotMove ()Z
-	METHOD method_6063 getHealthMaximum ()F
+	METHOD method_6061 canPierceDefense (Lnet/minecraft/class_1282;)Z
```

Comment:
This is wrong, it actually returns whether an attack hits the shield:

```java
    private boolean hitsShield(DamageSource source) {
        Entity attacker = source.getSource();
        boolean pierces = false;

        if (attacker instanceof ProjectileEntity) {
            ProjectileEntity projectile = (ProjectileEntity) attacker;
            if (projectile.getPierceLevel() > 0) {
                pierces = true;
            }
        }

        if (!source.bypassesArmor() && isBlocking() && !pierces) { // using shield and damage doesn't bypass armor
            Vec3d pos = source.getPosition();
            if (pos != null) {
                Vec3d rotation = getRotationVec(1.0F);
                Vec3d diff = pos.reverseSubtract(new Vec3d(x, y, z)).normalize();
                diff = new Vec3d(diff.x, 0.0D, diff.z);
                return diff.dotProduct(rotation) < 0.0D; // shield was hit
            }
        }

        return false;
    }
```

```java
            if (amount > 0.0F && hitsShield(source)) {
                damageShield(amount);
                damageAbsorbedByShield = amount;
                amount = 0.0F;
                if (!source.isProjectile()) {
                    Entity attacker = source.getSource();
                    if (attacker instanceof LivingEntity) {
                        takeShieldHit((LivingEntity) attacker);
                    }
                }

                shieldWasHit = true;
            }
```

Reply:
@Sollace You renamed this `tryUseShield`, but it doesn't actually use the shield, it just returns true if the shield can block the attack (if the player is blocking, and the line between the the entity and attacker intersects the shield). The second piece of code above shows the *usage* of the `hitsShield` method.

Reply:
@Runemoro `tryUseShield(source)` seems to work fine in that context, though.

```
        if (amount > 0 && tryUseShield(source)) {
            blockByShield = true;

            damageShield(amount);
            amountAbsorbedByShield = amount;
            amount = 0;

            if (!source.isProjectile()) {
               Entity attacker = source.getSource();
               if (attacker instanceof LivingEntity) {
                  takeShieldHit((LivingEntity)attacker);
               }
            }
         }
```

If the amount is >0 (not fully absorbed) it tries to use the shield and then does the things that only occur the shield succeeded in blocking the attack.

Reply:
> it tries to use the shield

No. It checks if the shield can block the attack. Trying to use the shield would mean that it's possible for a call to `tryUseShield` to actually use the shield and damage it, subtract the absorbed damage, play shield hit sound, etc.

Checking if you can use something is in no way the same as actually using (or attempting to use) something.

Reply:
Alright, changed.